planner/core/rule_topn_push_down.go

```diff
@@ -107,6 +107,12 @@ func (p *LogicalUnionAll) pushDownTopN(topN *LogicalTopN) LogicalPlan {
 }
 
 func (p *LogicalProjection) pushDownTopN(topN *LogicalTopN) LogicalPlan {
+	for _, expr := range p.Exprs {
+		if expression.HasAssignSetVarFunc(expr) {
+			topN.SetChildren(p)
+			return topN
+		}
+	}
```

Comment:
Sorry for the misleading before.
Try this:
```golang
if expression.HasAssignSetVarFunc(expr) {
    return p.baseLogicalPlan.pushDownTopN(topN)
}
```

Reply:
Fixed, PTAL.